Assets/MixedRealityToolkit.Tests/PlayModeTests/Core/ChangeActiveProfileTests.cs

```diff
@@ -0,0 +1,164 @@
+// Because the .NET backend is deprecated in 2018 and removed in 2019 and this
+// issue will likely persist for 2018, this issue is worked around by wrapping all
+// play mode tests in this check.
+
+using Microsoft.MixedReality.Toolkit.Boundary;
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using NUnit.Framework;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    // Tests to verify that changing the active profile at runtime results in
+    // the MRTK being correctly reconfigured.
+    public class ChangeActiveProfileTests
+    {
+        [TearDown]
+        public void TearDown()
+        {
+            // todo - cleanup scene
+
+            TestUtilities.ShutdownMixedRealityToolkit();
+        }
+
+        private const string CameraInputDiagsProfile = "Assets/MixedRealityToolkit.Tests/PlayModeTests/Core/TestProfiles/CameraInputDiags.asset";
+        private const string BoundaryOnlyProfile = "Assets/MixedRealityToolkit.Tests/PlayModeTests/Core/TestProfiles/BoundaryOnly.asset";
+
+        [UnityTest]
+        public IEnumerator VerifyServiceCount()
+        {
+            MixedRealityToolkitConfigurationProfile profile1 = TestUtilities.GetDefaultMixedRealityProfile<MixedRealityToolkitConfigurationProfile>();
+            yield return null;
+
+            // Initialize the test case with profile 1
+            InitializeTest(profile1);
+
+            // Get count of registered services
+            IReadOnlyList<IMixedRealityService> services = MixedRealityServiceRegistry.GetAllServices();
+            int count1 = services.Count;
+            yield return null;
+
+            // Switch to profile 2
+            MixedRealityToolkitConfigurationProfile profile2 = LoadTestProfile(CameraInputDiagsProfile);
+            ChangeProfile(profile2);
+            yield return null;
+
+            // Get count of registered services
+            services = MixedRealityServiceRegistry.GetAllServices();
+            int count2 = services.Count;
+
+            // Compare the service counts
+            Assert.IsTrue(count1 != count2);
+        }
+
+        [UnityTest]
+        public IEnumerator VerifyServiceState()
+        {
+            MixedRealityToolkitConfigurationProfile profile1 = TestUtilities.GetDefaultMixedRealityProfile<MixedRealityToolkitConfigurationProfile>();
+            yield return null;
+
+            // Initialize the test case with profile 1
+            InitializeTest(profile1);
+
+            // Cache the interesting settings read from the initial boundary system instance
+            IMixedRealityBoundarySystem boundarySystem1 = null;
+            MixedRealityServiceRegistry.TryGetService<IMixedRealityBoundarySystem>(out boundarySystem1);
```

Comment:
A lot of our other code doesn't use the registry directly now and just goes like:

https://microsoft.github.io/MixedRealityToolkit-Unity/version/releases/2.2.0/Documentation/Boundary/BoundarySystemGettingStarted.html#accessing-boundary-system-via-code

i.e. CoreServices.BoundaryService

To reduce the verbosity and take advantage of caching stuff.

It's also interesting to note that, the values returned by CoreServices should ideally also change after profile switches

Reply:
@Troy-Ferrell FYI, I assume that we should probably access things that's consistent with how our stuff is documented

Reply:
Good call (old habits die hard). I will update the test.

Reply:
fixed